src/context/EarnManagerProvider/providers/yearn/components/YearnManager/YearnWithdraw.tsx

```diff
@@ -1,67 +1,149 @@
-  { label: 'Confirm' },
-  { label: 'Broadcast' }
+enum WithdrawPath {
+  Withdraw = '/withdraw',
+  Confirm = '/withdraw/confirm',
+  ConfirmSettings = '/withdraw/confirm/settings',
+  Broadcast = '/withdraw/broadcast'
+}
+
+export const routes = [
+  { step: 0, path: WithdrawPath.Withdraw, label: 'Withdraw Amount' },
+  { step: 1, path: WithdrawPath.Confirm, label: 'Confirm Withdraw' },
+  { path: WithdrawPath.ConfirmSettings, label: 'Confirm Settings' },
+  { step: 2, path: WithdrawPath.Broadcast, label: 'Broadcast' }
 ]
 
-export const YearnWithdraw = ({ api }: { api: YearnVaultApi }) => {
-  const history = useHistory()
-  const { nextStep, activeStep } = useSteps({ initialStep: 0 })
+type YearnWithdrawProps = {
+  api: YearnVaultApi
+}
+
+export const YearnWithdraw = ({ api }: YearnWithdrawProps) => {
+  const [apy, setApy] = useState('0')
+  const [userAddress, setUserAddress] = useState<string | null>(null)
+  const [, /* values */ setValues] = useState<WithdrawValues>({} as WithdrawValues)
+  const { query, history: browserHistory } = useBrowserRouter<EarnQueryParams, EarnParams>()
+  const { chain, contractAddress: vaultAddress, tokenId } = query
+
+  // Asset info
+  const asset = useFetchAsset({ chain, tokenId })
+  const marketData = useMarketData({ chain, tokenId })
 
-  const handlePercentChange = () => {}
+  // user info
+  const chainAdapterManager = useChainAdapters()
+  const { state: walletState } = useWallet()
+  const { balances, loading } = useFlattenedBalances()
 
-  const handleContinueWithdraw = () => {
-    nextStep()
+  // navigation
+  const memoryHistory = useHistory()
+  const location = useLocation()
+  const WithdrawRoute = matchPath(location.pathname, { path: WithdrawPath.Withdraw, exact: true })
+
+  useEffect(() => {
+    ;(async () => {
+      if (!walletState.wallet) return
+      const chainAdapter = chainAdapterManager.byChain(ChainTypes.Ethereum)
+      setUserAddress(await chainAdapter.getAddress({ wallet: walletState.wallet }))
+    })()
+  }, [chain, chainAdapterManager, walletState])
+
+  useEffect(() => {
+    ;(async () => {
+      try {
+        const apy = await api.apy({ vaultAddress })
+        setApy(apy)
+      } catch (error) {
+        // TODO: handle client side errors
+        console.error('error', error)
+      }
+    })()
+  }, [api, vaultAddress])
+
+  const handleContinue = async (formValues: WithdrawValues) => {
+    setValues(formValues)
+    memoryHistory.push(WithdrawPath.Confirm)
   }
 
   const handleConfirm = async () => {
-    nextStep()
+    memoryHistory.push(WithdrawPath.Broadcast)
   }
 
+  const handleViewPosition = () => {}
+
   const handleCancel = () => {
-    history.goBack()
+    browserHistory.goBack()
   }
 
-  const handleViewAsset = () => {
-    console.info('view asset')
+  const balance = balances[tokenId ?? chain]?.balance
+
+  const validateCryptoAmount = (value: string) => {
+    const crypto = bnOrZero(balance).div(`1e+${asset.precision}`)
```

Comment:
Do we need to verify that `asset.precision` exists? (If the answer is "no", that's fine and no changes needed).

Reply:
No, precision is required on the asset type.